uaclient/config.py

```diff
@@ -78,7 +78,6 @@ class UAConfig:
     data_paths = {
```

Comment:
This is just an idea, but let me know what you think about this. What if we treat every file in `DataPath` as a `FileObject`.
We could have a base `FileObject`. That object would be the one that have the methods for `write_cache`  and `read_cache`. If we do this, we can make our config `write_cache` and `read_cache` just a matter of identifying the right `FileObject` and calling the appropriate method there.

We could have three distinct classes for the `machine_token` file: `MachineTokenPrivateFile`, `MachineTokenPublicFile` and `MachineTokenFile`. Both private and public classes would just inherit from the base `FileObject` class. However, the public file will just have a way to redact the content before writing it. Additionally, the `MachineTokenFile` would just need to handle which type of object to return, the public or the private one.

Why would that be useful ? This would make our `write_cache`, `read_cache` and `delete_cache` complete agnostic to the type of `FileObject` instance being used, we would not need to differentiate anymore from a normal file from a `MachineTokenFile` as we are doing in the code.

For example, we would turn `cfg.write_cache` into:
```python
def write_cache(key, content):
    file_obj = self.get_file_object(key)
    file_obj.write(content)
```

Same structure for `delete_cache_key`  and we can loop on the object for `delete_cache`.
This idea would also allow to have less code duplication between `UAConfig` and `MachineTokenFile`

@orndorffgrant @renanrodrigo @CalvoM Let me know if this makes sense or if I should better detail this idea.

Reply:
This is a good idea. We will just have to deal with the unsetting of the machine_token and entitlements for the machine-token file.

Reply:
Yes this is the direction I'd like to head in. My long term vision is:

1. Each file gets a `FileObject` like you describe
2. Internal to each `FileObject`, the data from the file is modeled as a `DataObject` rather than a `dict`
3. Remove all file storage from the UAConfig class.
4. A new singleton class called `Store` or something like that, that keeps a lazy loaded reference to each available `FileObject`

But for now (as agreed just now on a call), let's define the scope of this PR a bit.

This PR should include:
* The new `FileObject` abstraction that we can reuse for all our storage/state files (but dont refactor any existing file reading/writing)
* The new `MachineTokenPrivate`, `MachineTokenPublic` FileObjects
* The new `MachineToken` that abstracts over both public and private machine token files